ui/appui-react/src/appui-react/ui-items-provider/AbstractUiItemsManager.ts

```diff
@@ -0,0 +1,548 @@
+    const abstractItems = items.map((item) => toAbstractBackstageItem(item));
+    return abstractItems;
+  }
+
+  public provideStatusBarItems(stageId: string, stageUsage: string, _stageAppData?: any) {
+    if (!this._adaptee.provideStatusBarItems)
+      return [];
+
+    const items = this._adaptee.provideStatusBarItems(stageId, stageUsage);
+    const abstractItems = items.map((item) => toAbstractStatusBarItem(item));
+    return abstractItems;
+  }
+
+  public provideToolbarButtonItems(stageId: string, stageUsage: string, usage: AbstractToolbarUsage, orientation: AbstractToolbarOrientation, _stageAppData?: any) {
+    if (!this._adaptee.provideToolbarItems)
+      return [];
+
+    const items = this._adaptee.provideToolbarItems(stageId, stageUsage, usage, orientation);
+    const abstractItems = items.map((item) => toAbstractToolbarItem(item));
+    return abstractItems;
+  }
+
+  public provideWidgets(stageId: string, stageUsage: string, abstractLocation: AbstractStagePanelLocation, abstractSection?: AbstractStagePanelSection | undefined, _zoneLocation?: any, _stageAppData?: any) {
+    if (!this._adaptee.provideWidgets)
+      return [];
+
+    const location = fromAbstractStagePanelLocation(abstractLocation);
+    const section = abstractSection === undefined ? undefined : fromAbstractStagePanelSection(abstractSection);
+    const widgets = this._adaptee.provideWidgets(stageId, stageUsage, location, section);
+    const abstractWidgets = widgets.map((widget) => toAbstractWidget(widget));
+    return abstractWidgets;
+  }
+}
+
+class AbstractToUiItemsProviderAdapter implements UiItemsProvider {
+  public get id(): string {
+    return this._adaptee.id;
+  }
+
+  constructor(private readonly _adaptee: AbstractUiItemsProvider) {
+  }
+
+  public onUnregister() {
+    this._adaptee.onUnregister?.();
+  }
+
+  public provideBackstageItems() {
+    if (!this._adaptee.provideBackstageItems)
+      return [];
+
+    const abstractItems = this._adaptee.provideBackstageItems();
+    const items = abstractItems.map((abstractItem) => fromAbstractBackstageItem(abstractItem));
+    return items;
+  }
+
+  public provideStatusBarItems(stageId: string, stageUsage: string) {
+    if (!this._adaptee.provideStatusBarItems)
+      return [];
+
+    const abstractItems = this._adaptee.provideStatusBarItems(stageId, stageUsage);
+    const items = abstractItems.map((abstractItem) => fromAbstractStatusBarItem(abstractItem));
+    return items;
+  }
+
+  public provideWidgets(stageId: string, stageUsage: string, location: StagePanelLocation, section?: StagePanelSection) {
+    if (!this._adaptee.provideWidgets)
+      return [];
+
+    const abstractLocation = toAbstractStagePanelLocation(location);
+    const abstractSection = section === undefined ? undefined : toAbstractStagePanelSection(section);
+    const abstractWidgets = this._adaptee.provideWidgets(stageId, stageUsage, abstractLocation, abstractSection);
```

Comment:
Similar to the UiItemsManagerAdapter, if the Start section is requested from this adapter, it would need to also return the Middle section. (Although, I'm not sure I would understand the point of pulling out a provider and then call back its getter functions, this is what this try to provide)

Same thing for location.

Reply:
This is similar to Location, so the same argument applies.

Reply:
This is actually different, because this class is adapting an AbstractUiItemsProvider (which might only return values if `Middle` is requested) to a UiItemsProvider, which will never be requested to return such widget.

Reply:
I'm sorry, this was similar to `AbstractUiItemsManagerAdapterm`, fixed.